internal/api/httpsrv/handler_issuer.go

```diff
@@ -81,12 +97,23 @@ func (h *apiHandler) GetIssuerByID(ctx context.Context, req GetIssuerByIDRequest
 }
 
 func (h *apiHandler) UpdateIssuer(ctx context.Context, req UpdateIssuerRequestObject) (UpdateIssuerResponseObject, error) {
+	// We must fetch the issuer to retrieve the owner so we may check for permission to update.
+	iss, err := h.engine.GetIssuerByID(ctx, req.Id)
+	switch err {
+	case nil:
+	case types.ErrorIssuerNotFound:
+		return nil, errorNotFound
+	default:
+		return nil, err
+	}
+
+	if err := permissions.CheckAccess(ctx, iss.OwnerID, actionIssuerUpdate); err != nil {
```

Comment:
If something goes wrong checking access, this will return a 403. It seems like it should return a 500 if something breaks rather than permissions checks failing.

This can probably be fixed in the `permissionsError` function rather than in every conditional where we do this check.